### PR TITLE
Fix tox and add GH workflow

### PR DESCRIPTION
**Pull request by mikemanger**
Some fixes to the tests
- Fix the tox config (previously the fixtures weren't getting included so it would fail)
- Use assetEqual instead of assertEquals
- Remove unsupported nosetests file
- adds a GitHub workflow

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,17 +1,26 @@
-# Tox (http://tox.testrun.org/) is a tool for running tests
+# Tox (https://tox.wiki/) is a tool for running tests
 # in multiple virtualenvs. This configuration file will run the
 # test suite on all supported python versions. To use it, "pip install tox"
 # and then run "tox" from this directory.
 
 [tox]
-# pip is broken on py32
-envlist = py36, py37, py38, py39, py310, py311, py312, py313
+env_list = py36, py37, py38, py39, py310, py311, py312, py313
```
Comment (russella-acm) [inline tox.ini:7]: More dropping of py36 and py37:
```suggestion
env_list = py38, py39, py310, py311, py312, py313
```

File: tox.ini [hunk 2]
```diff
@@ -1,17 +1,26 @@
...
-    coverage
-    -e.
+    pytest-cov
+# This needs to be set to include the test fixtures
+use_develop = true
+
+[gh]
+python =
+    3.6: py36
+    3.7: py37
```
Comment (russella-acm) [inline tox.ini:20]: Final spot to consider dropping 3.6 and 3.7 here.
```suggestion
```

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  - push
+  - pull_request
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # Python 3.6 is not supported by ubuntu-latest
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11', '3.12', '3.13']
```
Comment (russella-acm) [inline .github/workflows/tests.yml:13]: Could we remove the comment about '3.6' as well as version '3.7'. The README can be updated to drop saying that we support versions 3.6 and 3.7 as well which I am happy to do after this PR is merged and at the time of release.
```suggestion
        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12', '3.13']
```

**Review by russella-acm — COMMENTED** [review]
Hi @mikemanger - thank you enormously for the excellent changes here in this pull request - we really appreciate your work here. I am eager to get this PR moving ahead.

I had a few things which I wanted to go over with you to get us to that stage. With all this, please keep in mind I am not an experienced Python developer and so there might be some things I am off the mark about and I'm happy to learn and be corrected along the way.

These were the items I had in mind:
* Earlier today, I merged in your work from https://github.com/campaignmonitor/createsend-python/pull/84 which have some changes in common with this PR - namely changing `assertEquals` to `assertEqual`. This is now in master and in the latest version of the package (8.0.2) and hopefully bringing those changes across in to this PR will reduce the number of changes.
* I am not too familiar with the github workflow side of things but I am happy to go ahead with those changes for the automated testing side of things.
* I ran `tox` and that was showing some issues with version Python 3.7, namely this issue shown below. Python version 3.6 is skipped and 3.7 doesn't run properly:
image

The error for Python 7 I was seeing on my machine was this:
image

How would we feel about removing Python versions 3.6 and 3.7 from `tests.yml` and `tox.ini`? I have left these as things to consider below. I'm happy to update the documentation saying that we do **not** support those two versions any more. Having said that, I did test with Python 3.6 and the wrapper appears to still be working as before.

* From Python version 3.11 and onwards, I noticed this deprecation warning that came up:
```
\createsend-python\lib\createsend\utils.py:106: DeprecationWarning: ssl.SSLContext() without protocol argument is deprecated.
    context = ssl.SSLContext()

\createsend-python\lib\createsend\utils.py:106: DeprecationWarning: ssl.PROTOCOL_TLS is deprecated
    context = ssl.SSLContext()
```

I attempted to remove this warning by changing utils.py line 106:
```
From:
ssl.SSLContext()

To:
ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
```

But that resulted in this failure:
image

```
FAILED test/test_verifiedhttpsconnection.py::VerifiedHTTPSConnectionTestCase::test_verified_connection - ValueError: check_hostname requires server_hostname
```

For now, I was thinking we're happy to leave it as a warning and figure out a fix for that separately.

So with all that in mind after:
* this PR has the latest from master merged in; and
* if you agree that removing 3.6 and 3.7 is a good idea and we do so

I'm happy to get this approved, merged and released ASAP. Thank you once again for your help and pushing for this code base to be modernised.

**Comment by mikemanger** [comment]
Hey @russella-acm  thanks for the great review!

Interestingly the python 3.7 tests runs fine on my GH workflow ([you can see this here](https://github.com/mikemanger/createsend-python/actions/runs/12196658868/job/34024692024)) so I'm wondering if it is an issue with pip on Windows (a quick search didn't return anything).

Either way I'm happy to drop support for Python 3.7 (its EOL over a year now). I've updated this branch and also fixed a couple of small typos.

With the 3.11 deprecation warning I think we should tackle that in another PR as important but outside the scope of this PR.

**Review by russella-acm — APPROVED** [review]
LGTM - Thank you @mikemanger - really appreciate the changes, I will go about getting these in and releasing them in a next version ASAP.

I tested out tox, unit tests and ran then the wrapper code. All working as expected.

**Pull request merged**